creator-node/src/tracer.ts

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * Initializes a tracer for content node as well as registers import instrumentions
+ * for packages that are frequently used
+ */
+export const setupTracing = () => {
+  /**
+   * A Tracer Provider is a factory for Tracers.
+   * A Tracer Provider is initialized once and its lifecycle matches the application’s lifecycle.
+   * Tracer Provider initialization also includes Resource and Exporter initialization.
+   * It is typically the first step in tracing with OpenTelemetry.
+   */
+  const provider = new NodeTracerProvider({
+    resource: new Resource({
+      [ResourceAttributesSC.SERVICE_NAME]: SERVICE_NAME
+    })
+  })
+  registerInstrumentations({
+    tracerProvider: provider,
+    instrumentations: [
+      // Express instrumentation expects HTTP layer to be instrumented
+      // This reads and writes the appropriate opentelemetry headers to requests
+      new HttpInstrumentation(),
+
+      // Adds spans to express routes
+      new ExpressInstrumentation(),
+
+      // Adds spans to redis operations
+      new RedisInstrumentation(),
+
+      // Adds spans to postgres transactions
+      new PgInstrumentation(),
+
+      // Injects traceid, spanid, and SpanContext into bunyan logs
+      new BunyanInstrumentation({
+        // Adds a hook to logs that injects more span info
+        // and the service name into logs
+        logHook: (span, record) => {
+          record['resource.span'] = span
+          record['resource.service.name'] =
+            provider.resource.attributes['service.name']
+        }
+      })
+    ]
+  })
+
+  // Initialize the OpenTelemetry APIs to use the NodeTracerProvider bindings
+  provider.register()
+}
+
+/**
+ * Namespace to store tracing helper function
+ */
+export const tracing = {
+  CODE_FILEPATH: SemanticAttributes.CODE_FILEPATH,
+  CODE_FUNCTION: SemanticAttributes.CODE_FUNCTION,
+  SpanStatusCode: SpanStatusCode,
+
+  /**
+   * A Tracer creates spans containing more information about what is happening
+   * for a given operation, such as a request in a service.
+   * Tracers are created from Tracer Providers.
+   *
+   * This function fetches the tracer from the application context
+   * @returns {Tracer} the tracer for content node
+   */
+  getTracer: () => {
+    return trace.getTracer(SERVICE_NAME)
+  },
+
+  /**
+   * Helper function that gets the current active span or return `undefined` if there is no active span
+   */
+  getActiveSpan: (): Span | undefined => {
+    return trace.getSpan(context.active())
+  },
+
+  setSpanAttribute: (name: string, value: AttributeValue) => {
+    const span = tracing.getActiveSpan()
+    span?.setAttribute(name, value)
+  },
+
+  info: (msg: string) => {
+    const span = tracing.getActiveSpan()
+    span?.addEvent(msg, {
+      'log.severity': 'info'
+    })
+  },
+
+  warn: (msg: string) => {
+    const span = tracing.getActiveSpan()
+    span?.addEvent(msg, {
+      'log.severity': 'warn'
+    })
+  },
+
+  error: (msg: string) => {
```

Comment:
whats purpose of the `info / warn / error` functions here? are they manually called on a span and set this metadata?

Reply:
Yes, they'll make it easier to add metadata to spans in the subsequent PRs. Other than `tracer.ts` no other file in CN should need to import any opentelemetry stuff

Reply:
makes sense - i just meant how are these consumed, but sg will see later

Reply:
Ya it'll make more sense later (they're really just in this PR because I copied the whole file over)

They'll get used just like a logger essentially

```js
try {
    tracing.info('blah blah blah')
    tracing.warn('something')
} catch (e) {
    tracing.recordException(e)
    tracing.error('something very bad happened')
}
```

Reply:
dope